enhance DDPSink to work for general outputs

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57081 [DDP] Support not all outputs used in loss calculation
* **#57073 enhance DDPSink to work for general outputs**

Enhances use of DDPSink to work for all output types DDP supports as per https://github.com/pytorch/pytorch/issues/55876.

TODO: Add additional testing for tuple, list, dict return types

Differential Revision: [D27756985](https://our.internmc.facebook.com/intern/diff/D27756985/)

rohan-varma committed:
[wip] enhance DDPSink to work for general outputs

Enhances use of DDPSink to work for all output types DDP supports as per https://github.com/pytorch/pytorch/issues/55876.

TODO: Add additional testing for tuple, list, dict return types

Differential Revision: [D27756985](https://our.internmc.facebook.com/intern/diff/D27756985/)

[ghstack-poisoned]

rohan-varma commented:
cc @zhaojuanmao Added tests for different DDP output types (list, tuple, dict) to verify that they work now with the `_DDPSink`.

rohan-varma committed:
Update on "[wip] enhance DDPSink to work for general outputs"


Enhances use of DDPSink to work for all output types DDP supports as per https://github.com/pytorch/pytorch/issues/55876.

TODO: Add additional testing for tuple, list, dict return types

Differential Revision: [D27756985](https://our.internmc.facebook.com/intern/diff/D27756985/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[wip] enhance DDPSink to work for general outputs"


Enhances use of DDPSink to work for all output types DDP supports as per https://github.com/pytorch/pytorch/issues/55876.

TODO: Add additional testing for tuple, list, dict return types

Differential Revision: [D27756985](https://our.internmc.facebook.com/intern/diff/D27756985/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[wip] enhance DDPSink to work for general outputs"


Enhances use of DDPSink to work for all output types DDP supports as per https://github.com/pytorch/pytorch/issues/55876.

TODO: Add additional testing for tuple, list, dict return types

Differential Revision: [D27756985](https://our.internmc.facebook.com/intern/diff/D27756985/)

[ghstack-poisoned]

zhaojuanmao reviewed (commented):
awesome, thanks, @rohan-varma!

rohan-varma committed:
Update on "[wip] enhance DDPSink to work for general outputs"


Enhances use of DDPSink to work for all output types DDP supports as per https://github.com/pytorch/pytorch/issues/55876.

TODO: Add additional testing for tuple, list, dict return types

Differential Revision: [D27756985](https://our.internmc.facebook.com/intern/diff/D27756985/)

[ghstack-poisoned]

zhaojuanmao reviewed (approved):
lgtm!

rohan-varma committed:
Update on "[wip] enhance DDPSink to work for general outputs"


Enhances use of DDPSink to work for all output types DDP supports as per https://github.com/pytorch/pytorch/issues/55876.

TODO: Add additional testing for tuple, list, dict return types

Differential Revision: [D27756985](https://our.internmc.facebook.com/intern/diff/D27756985/)

[ghstack-poisoned]

rohan-varma committed:
Update on "enhance DDPSink to work for general outputs"


Enhances use of DDPSink to work for all output types DDP supports as per https://github.com/pytorch/pytorch/issues/55876.

TODO: Add additional testing for tuple, list, dict return types

Differential Revision: [D27756985](https://our.internmc.facebook.com/intern/diff/D27756985/)

[ghstack-poisoned]
